src/Microsoft.Diagnostics.NETCore.Client/DiagnosticsIpc/ProcessInfo.cs

```diff
@@ -0,0 +1,69 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Text;
+
+namespace Microsoft.Diagnostics.NETCore.Client
+{
+    /**
+     * ==ProcessInfo==
+     * The response payload to issuing the GetProcessInfo command.
+     * 
+     * 8 bytes  - PID (little-endian)
+     * 16 bytes - CLR Runtime Instance Cookie (little-endian)
+     * # bytes  - Command line string length and data
+     * # bytes  - Operating system string length and data
+     * # bytes  - Process architecture string length and data
+     * 
+     * The "string length and data" fields are variable length:
+     * 4 bytes            - Length of string data in UTF-16 characters
+     * (2 * length) bytes - The data of the string encoded using Unicode
+     *                      (includes null terminating character)
+     */
+
+    internal class ProcessInfo
+    {
+        private static readonly int GuidSizeInBytes = 16;
+
+        public static ProcessInfo Parse(byte[] payload)
+        {
+            ProcessInfo processInfo = new ProcessInfo();
+
+            int index = 0;
+            processInfo.ProcessId = BitConverter.ToUInt64(payload, index);
+            index += sizeof(UInt64);
+
+            byte[] cookieBuffer = new byte[GuidSizeInBytes];
+            Array.Copy(payload, index, cookieBuffer, 0, GuidSizeInBytes);
+            processInfo.RuntimeInstanceCookie = new Guid(cookieBuffer);
+            index += GuidSizeInBytes;
+
+            processInfo.CommandLine = ReadString(payload, ref index);
+            processInfo.OperatingSystem = ReadString(payload, ref index);
+            processInfo.ProcessArchitecture = ReadString(payload, ref index);
+
+            return processInfo;
+        }
+
+        private static string ReadString(byte[] buffer, ref int index)
```

Comment:
nit: ReadOnlySpan

Reply:
That's not available in netstandard2.0 without using the System.Memory NuGet package. I'm not certain that we want to have NuGet dependencies for this library (there currently aren't any). Maybe someone from @dotnet/dotnet-diag can chime in on whether adding NuGet dependencies to Microsoft.Diagnostics.NETCore.Client is allowed, to be avoided, maybe some are okay, etc.

Reply:
I have no qualms with dependencies in principle and System.Memory seems fine to reference. If some other potential dependency was larger or we didn't have the power to service bugs those would be some reasons I'd more hesitant.

Reply:
I'm going to defer making this change at this time. There is opportunity to use System.Memory elsewhere in this library and I think it would be better to make one cohesive change to make use of it rather than as individual changes.